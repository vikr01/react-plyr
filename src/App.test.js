import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Plyr from './index';

it('renders a simple Youtube Video', () => {
  const wrapper = shallow(
    <Plyr
      type="youtube"
      videoId="CDFN1VatiJA"
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('renders a simple Vimeo Video', () => {
  const wrapper = shallow(
    <Plyr
      type="vimeo"
      videoId="https://vimeo.com/189789787"
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('renders a simple HTML5 Video with Captions', ()=>{
  const wrapper = shallow(
    <Plyr
    type="video"
    poster="/path/to/poster.jpg"
    url="mymovie.mp4"
    captions={[
      {
        kind: 'captions',
        label: 'English captions',
        src: '/path/to/english-captions.vtt',
        srclang: 'en',
        default: true
      },
      {
        label: 'Spanish captions',
        src: '/path/to/spanish-captions.vtt',
        srclang: 'es',
      }
    ]}
    />
  )
  expect(toJson(wrapper, {noKey: true, mode: 'deep'})).toMatchSnapshot();
});