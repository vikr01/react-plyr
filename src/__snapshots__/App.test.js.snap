// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a simple HTML5 Video with Captions 1`] = `
<video
  className="react-plyr"
  poster="/path/to/poster.jpg"
  src="mymovie.mp4"
>
  <track
    default={true}
    kind="captions"
    label="English captions"
    src="/path/to/english-captions.vtt"
    srclang="en"
  />
  <track
    kind="captions"
    label="Spanish captions"
    src="/path/to/spanish-captions.vtt"
    srclang="es"
  />
</video>
`;

exports[`renders a simple Vimeo Video 1`] = `
<div
  className="react-plyr"
  data-plyr-embed-id="https://vimeo.com/189789787"
  data-plyr-provider="vimeo"
/>
`;

exports[`renders a simple Youtube Video 1`] = `
<div
  className="react-plyr"
  data-plyr-embed-id="CDFN1VatiJA"
  data-plyr-provider="youtube"
/>
`;
